File: hello-world/hello.c.rb
#if 0 // code visible to Ruby

def sayHello
  puts("Hello from \x1b[91mRuby\x1b[m!")
end

=begin code visible to C
#endif

#ifndef __COSMOPOLITAN__
#include <stdio.h>
#endif
#define END int main()

void sayHello() {
  puts("Hello from \x1b[90mC\x1b[m!");
}

#if 0
=end
#endif // code visible to C and Ruby

END {
  sayHello();
  puts("Hello from \x1b[90mC\x1b[m and \x1b[91mRuby\x1b[m!");
}

File: hello-world/hello_c.js
//\
/* Code visible to C

#ifndef __COSMOPOLITAN__
#include <stdio.h>
#include <stdlib.h>
#endif
#define function int
#define print puts

void sayHello() {
  print("Hello from \x1b[90mC\x1b[m!");
}

/*/ // Code visible to JavaScript

var print;
function sayHello() {
  if (print) {
    print("Hello from \x1b[33mqjs\x1b[m!");
  } else if (this == undefined) {
    print = console.log;
    print("Hello from \x1b[33mdeno\x1b[m!");
  } else if (this.display) {
    this.console = { log: print = function(s) { display(s); newline(); }};
    print("Hello from \x1b[33mguile\x1b[m!");
  } else {
    print = console.log;
    print("Hello from \x1b[33mnode\x1b[m!");
  }
}

// Code visible to C and JavaScript */

function main() {
  sayHello();
  print("Hello from \x1b[90mC\x1b[m and \x1b[33mJavaScript\x1b[m!");
}

//\
main();
